components/popover/popover-mixin.js

```diff
@@ -127,68 +215,375 @@ export const PopoverMixin = superclass => class extends superclass {
+			constrained.right = Math.max(0, spaceRequired.width - (openerRect.width + spaceAround.left));
+		}
+
+		return constrained;
+	}
+
+	#focusContent(container) {
 		if (this._noAutoFocus || this._applyFocus === false) return;
 
 		const focusable = getFirstFocusableDescendant(container);
 		if (focusable) {
-			// Removing the rAF call can allow infinite focus looping to happen in content using a focus trap
+			// removing the rAF call can allow infinite focus looping to happen in content using a focus trap
 			requestAnimationFrame(() => focusable.focus());
 		} else {
-			const content = this._getContentContainer();
+			const content = this.#getContentContainer();
 			content.setAttribute('tabindex', '-1');
 			content.focus();
 		}
 	}
 
-	_focusOpener() {
+	#focusOpener() {
 		if (!document.activeElement) return;
 		if (!isComposedAncestor(this, getComposedActiveElement())) return;
 
 		this?._opener.focus();
 	}
 
-	_getContentContainer() {
-		return this.shadowRoot.querySelector('.content');
+	#getContentContainer() {
+		return this.shadowRoot.querySelector('.content-container');
+	}
+
+	#getLocation(spaceAround, spaceAroundScroll, spaceRequired) {
+
+		const preferred = this._preferredPosition;
+		if (!preferred.allowFlip) {
+			return preferred.location;
+		}
+
+		if (preferred.location === 'block-end') {
+			if (spaceAround.below >= spaceRequired.height) return 'block-end';
+			if (spaceAround.above >= spaceRequired.height) return 'block-start';
+			// if auto-fit is enabled, scroll will be enabled for the inner content so it will always fit in the available space so pick the largest space it can be displayed in
+			if (!this.noAutoFit) return spaceAround.above > spaceAround.below ? 'block-start' : 'block-end';
+			if (spaceAroundScroll.below >= spaceRequired.height) return 'block-end';
+			if (spaceAroundScroll.above >= spaceRequired.height) return 'block-start';
+		}
+
+		if (preferred.location === 'block-start') {
+			if (spaceAround.above >= spaceRequired.height) return 'block-start';
+			if (spaceAround.below >= spaceRequired.height) return 'block-end';
+			// if auto-fit is enabled, scroll will be enabled for the inner content so it will always fit in the available space so pick the largest space it can be displayed in
+			if (!this.noAutoFit) return spaceAround.above > spaceAround.below ? 'block-start' : 'block-end';
+			if (spaceAroundScroll.above >= spaceRequired.height) return 'block-start';
+			if (spaceAroundScroll.below >= spaceRequired.height) return 'block-end';
+		}
+
+		// todo: add location order for inline-start and inline-end
+
+		// if auto-fit is disabled and it doesn't fit in the scrollable space above or below, always open down because it can add scrollable space
+		return 'block-end';
+	}
+
+	#getPointer() {
+		return this.shadowRoot.querySelector('.pointer');
+	}
+
+	#getPointerPosition(openerRect) {
```

Comment:
This logic is almost exactly the same as `DropdownContentMixin` but I adjusted the calculation for the RTL `align`/`span` cases since I noticed they were not working quite right.